### DISCUSSION TITLE
Backpack under cursor

### DISCUSSION THREAD

**Discussion by nheath** [P3pp3rF1y/SophisticatedBackpacks#148] [Ideas]
I like to play as a nomad, without setting up a home, and this mod is great supporting that play style. I sometimes have multiple backpacks, and would like to interact with a specific backpack in my inventory. 

From the inventory/backpack screen, Is there a way to either open a specific backpack (Without putting it on the hotbar and right clicking)? A suggestion: when pressing the backpack key, in the inventory/backpack screens, see if the item under the mouse cursor is a backpack, and if so, open that backpacks inventory.

From the inventory/backpack screen, is there a way to insert an item into a specific backpack? i.e. in my player inventory, can i pick up and insert an item into a backpack without opening that backpacks inventory? A suggestion: pick up an item (or stack) and shift click a backpack in your inventory to insert the item into that backpack.

Thanks for considering these usability improvements!

**Comment by RGFTheCoder** [comment]
I agree, this would be an extremely fun feature to use

**Comment by P3pp3rF1y** [comment]
I like this and will see how easy it would be to implement. The open any backpack with keybind should hopefully be pretty easy. 
Clicking items directly into backpacks probably not so much. I may be able to do something about backpack, player's inventory or chest inventory here, but I am pretty sure I can't make it generic so that it would work with any inventory as if I remember this part of interaction correctly it would require injecting code in that mod for it to work.

As a side note you can already open backpacks in backpack if you have an inception upgrade so there's that.

**Comment by TheDeviantCrafter** [comment]
Maybe add an upgrade that allows backpacks to be opened by right clicking in the inventory, assuming your curser isn't holding an item. This could also be a property of the inception upgrade and only work with nested backpacks.

With Quark, picking up an item in the inventory and right clicking on a shulker box stores the item inside. Quark is open source.

**Reply by P3pp3rF1y** [reply]
I feel that this should be just a function of the backpack with no additional upgrade required. 

Thanks for pointing out that Quark does it, I have seen the functionality but wasn't sure what mod that was. Based on their implementation it actually looks pretty simple so I should be able to add this and it almost looks like it should be able to work in any inventory regardless if vanilla or modded. (of course now that I said this it's bound to be broken in some modded inventory)
